ipv8/peerdiscovery/network.py

```diff
@@ -273,7 +274,6 @@ def remove_by_address(self, address: Address) -> None:
         """
         with self.graph_lock:
             self._all_addresses.pop(address, None)
-            self.reverse_service_lookup.pop(address, None)
```

Comment:
Can you explain why this line can be removed? (as well as the following similar line)

Reply:
Sure: `reverse_service_lookup` is a `Cache of service_id -> [Peer]` (see line 57) and `address` is of type `Address` (the same is true for the next line `Peer.address`). This will never fail but it will also never do anything useful because addresses are not part of the `reverse_service_lookup`. Therefore, it has been removed.